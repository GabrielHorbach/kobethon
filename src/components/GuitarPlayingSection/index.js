import React from 'react';

import GuitarPlaying from '../../assets/animations/playing_guitar.gif';

import Container from './styles';

function GuitarPlayingSection() {
  return (
    <Container>
      <img src={GuitarPlaying} alt="Guitar playing gif" />

      <div>
        <h2>Hobbies</h2>
        <p>Bom, definitivamente <strong>música</strong> é o maior deles.</p>
        <p><br />
          Desde muito pequeno amo música e ao longo dos anos aprendi diversos instrumentos e canto,
          sendo violão e guitarra meus favoritos.
        </p><br />
        <p>Sempre tive mais facilidade com instrumentos de corda, mas isso não impediu que eu arranhasse algo no piano, saxofone e bateria também.</p>
      </div>
    </Container>
  );
}

export default GuitarPlayingSection;
