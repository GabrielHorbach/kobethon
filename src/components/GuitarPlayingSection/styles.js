import styled from 'styled-components';

const Container = styled.section`
  display: flex;
  height: 100vh;
  padding: 0 8rem;
  align-items: center;
  background: #f1f1f1;
  position: relative;

  @media (max-width: 1200px) {
    padding: 0 3rem;
  }

  @media (max-width: 650px) {
    flex-direction: column;
    height: auto;
    padding-bottom: 10%;
  }

  img {
    width: 50%;
    object-fit: contain;
    margin-right: 4rem;
    margin-left: -5rem;
    height: 70%;

    @media (max-width: 900px) {
      margin-left: -1rem;
      margin-right: 3rem;
      width: 40%;
    }

    @media (max-width: 650px) {
      width: 25%;
      height: auto;
      margin: 10rem 0 1rem;
    }
  }

  div {
    width: 50%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    border-radius: 15px;
    height: auto;
    margin-top: 2.5rem;

    @media (max-width: 650px) {
      margin-top: 0;
      width: 100%;
    }

    h2 {
      text-transform: uppercase;
      font-weight: bold;
      font-size: 3.5rem;
      position: absolute;
      top: 10%;
      color: #4E5B82;

      @media (max-width: 900px) {
        font-size: 2.5rem;
        top: 7%;
        left: 50%;
        transform: translateX(-50%);
      }
    }

    p {
      color: #4E5B82;
      font-size: 1.5rem;
      line-height: 2.5rem;

      @media (max-width: 1100px) {
        font-size: 1.3rem;
        line-height: 2.3rem;
      }

      @media (max-width: 800px) {
        font-size: 1.2rem;
        line-height: 2.2rem;
      }
    }
  }
`;

export default Container;
