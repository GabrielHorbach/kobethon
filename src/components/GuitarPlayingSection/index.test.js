import React from 'react';

import { shallow } from 'enzyme';

import GuitarPlayingSection from './index';

describe('GuitarPlayingSection Section', () => {
  const wrapper = shallow(<GuitarPlayingSection />);

  it('should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render gif', () => {
    const gif = wrapper.find('img');
    expect(gif).toBeTruthy();
  });
});