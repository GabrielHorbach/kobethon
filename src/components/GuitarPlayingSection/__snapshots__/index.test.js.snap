// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuitarPlayingSection Section should render information block 1`] = `ShallowWrapper {}`;

exports[`GuitarPlayingSection Section should render without errors 1`] = `ShallowWrapper {}`;
